### PR TITLE
TFTRT: Fix bugs with Concat and clean up

### PR DESCRIPTION
**Pull request by trevor-m**
Previously, the Concat layer had a bug where we only checked if the first input was a tensor, and then assumed all N inputs were tensors as well. This could lead to segfaults. Now we also perform the necessary checks.

### CODE REVIEW

**Review by aaroey — CHANGES_REQUESTED** [review]
Thanks @trevor-m for the fix. I think `ops::Concat()` will result in a ConcatV2 op, so please help to add unit test using that.

**Comment by trevor-m** [comment]
> Thanks @trevor-m for the fix. I think `ops::Concat()` will result in a ConcatV2 op, so please help to add unit test using that.

Thanks Lambda. I will add some unit tests.

**Comment by trevor-m** [comment]
@aaroey I have added the unit tests.

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc [hunk 1]
```diff
@@ -3407,77 +3416,78 @@ tensorflow::Status ConvertPad(OpConverterParams* params) {
...
-
-    inputs_vec.push_back(tensor_i);
   }
   if (params->validation_only) return tensorflow::Status::OK();
 
-  // nvinfer1::ITensor const* tensor = inputs.at(0).tensor();
+  // Gather inputs as tensors
+  std::vector<nvinfer1::ITensor const*> input_tensors;
+  for (int i = 0; i < num_inputs; i++) {
+    input_tensors.push_back(params->converter->GetAsTensor(inputs.at(i)));
```
[the remark this answers is not in the document]
Comment (aaroey) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: Another problem is, as chatted, IConstantLayer will add implicit batch dims, how to handle that? Also please add corresponding tests. Thanks.

**Review by aaroey — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc [hunk 2]
```diff
@@ -367,6 +367,15 @@ nvinfer1::ITensor* Converter::CreateConstantLayer(
   return trt_tensor;
 }
 
+// TODO(tmorris): Use this in more ops where we allow want to allow weights or
+// tensors for an input.
+// TODO(tmorris): Add a caching mechanism so many constant layers are not
```
[the remark this answers is not in the document]
Comment (trevor-m) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: @aaroey How should we handle the possibility that a constant is created for a certain weight but with different dims? Do we need to add the dims as a second key to the cache?

**Comment by trevor-m** [comment]
@aaroey Could you review this when you have a chance? Thanks

**Review by aaroey — DISMISSED** [review]

**Review by aaroey — APPROVED** [review]

**Pull request merged**